开启 oidc 后无法启动

<!-- 这段文字不会显示在你的内容中。为了避免重复的信息，方便后续的检索，在提issue之前，请检查如下事项。如果是比较新手级别的问题，推荐到讨论区https://github.com/ctripcorp/apollo/discussions 提问 -->

- [x] 我已经检查过[discussions](https://github.com/ctripcorp/apollo/discussions)
- [x] 我已经搜索过[issues](https://github.com/ctripcorp/apollo/issues)
- [x] 我已经仔细检查过[FAQ](https://www.apolloconfig.com/#/zh/faq/common-issues-in-deployment-and-development-phase)

**描述bug**

使用 keycloak 15 做 oidc 认证

**复现**

通过如下步骤可以复现：

1. 使用 keycloak 15 做 oidc 认证

**期望**

正常启动

**截图**

![image](https://user-images.githubusercontent.com/13741751/156920003-98e6faa9-1178-4cca-a497-1b8cb3838971.png)

### 额外的细节和日志

- 版本：2.0
- 错误日志
- 	at org.springframework.security.oauth2.client.registration.ClientRegistrations.getBuilder(ClientRegistrations.java:221)
	at org.springframework.security.oauth2.client.registration.ClientRegistrations.fromIssuerLocation(ClientRegistrations.java:145)
	at org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientPropertiesRegistrationAdapter.getBuilderFromIssuerIfPossible(OAuth2ClientPropertiesRegistrationAdapter.java:83)
	at org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientPropertiesRegistrationAdapter.getClientRegistration(OAuth2ClientPropertiesRegistrationAdapter.java:59)
	at org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientPropertiesRegistrationAdapter.lambda$getClientRegistrations$0(OAuth2ClientPropertiesRegistrationAdapter.java:53)
	at java.base/java.util.HashMap.forEach(HashMap.java:1421)
	at org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientPropertiesRegistrationAdapter.getClientRegistrations(OAuth2ClientPropertiesRegistrationAdapter.java:52)
	at org.springframework.boot.autoconfigure.security.oauth2.client.servlet.OAuth2ClientRegistrationRepositoryConfiguration.clientRegistrationRepository(OAuth2ClientRegistrationRepositoryConfiguration.java:49)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
	... 35 common frames omitted
Caused by: java.lang.RuntimeException: com.nimbusds.oauth2.sdk.ParseException: Unexpected type of JSON object member with key mtls_endpoint_aliases
	at org.springframework.security.oauth2.client.registration.ClientRegistrations.parse(ClientRegistrations.java:232)
	at org.springframework.security.oauth2.client.registration.ClientRegistrations.lambda$oidc$0(ClientRegistrations.java:157)
	at org.springframework.security.oauth2.client.registration.ClientRegistrations.getBuilder(ClientRegistrations.java:209)
	... 47 common frames omitted
Caused by: com.nimbusds.oauth2.sdk.ParseException: Unexpected type of JSON object member with key mtls_endpoint_aliases
	at com.nimbusds.oauth2.sdk.util.JSONObjectUtils.getGeneric(JSONObjectUtils.java:161)
	at com.nimbusds.oauth2.sdk.util.JSONObjectUtils.getJSONObject(JSONObjectUtils.java:827)
	at com.nimbusds.oauth2.sdk.as.AuthorizationServerMetadata.parse(AuthorizationServerMetadata.java:2042)
	at com.nimbusds.openid.connect.sdk.op.OIDCProviderMetadata.parse(OIDCProviderMetadata.java:1362)
	at org.springframework.security.oauth2.client.registration.ClientRegistrations.parse(ClientRegistrations.java:229)
	... 49 common frames omitted
Caused by: com.nimbusds.oauth2.sdk.ParseException: Unexpected type: class java.util.LinkedHashMap
	at com.nimbusds.oauth2.sdk.util.JSONUtils.to(JSONUtils.java:100)
	at com.nimbusds.oauth2.sdk.util.JSONObjectUtils.getGeneric(JSONObjectUtils.java:159)
	... 53 common frames omitted
- 配置：
- 平台和操作系统

--- a/apollo-portal/src/main/config/application-oidc-sample.yml
+++ b/apollo-portal/src/main/config/application-oidc-sample.yml
@@ -22,19 +22,19 @@ spring:
       client:
         provider:
           # provider-name 是 oidc 提供者的名称, 任意字符均可, registration 的配置需要用到这个名称
-          provider-name:
+          <fill-in-the-provider-name-here>:
             # 必须是 https, oidc 的 issuer-uri, 和 jwt 的 issuer-uri 一致的话直接引用即可, 也可以单独设置
             issuer-uri: ${spring.security.oauth2.resourceserver.jwt.issuer-uri}
         registration:
           # registration-name 是 oidc 客户端的名称, 任意字符均可, oidc 登录必须配置一个 authorization_code 类型的 registration
-          registration-name:
+          <fill-in-the-registration-name-here>:
             # oidc 登录必须配置一个 authorization_code 类型的 registration
             authorization-grant-type: authorization_code
             client-authentication-method: basic
             # client-id 是在 oidc 提供者处配置的客户端ID, 用于登录 provider
             client-id: apollo-portal
             # provider 的名称, 需要和上面配置的 provider 名称保持一致
-            provider: provider-name
+            provider: <fill-in-the-provider-name-here>
             # openid 为 oidc 登录的必须 scope, 此处可以添加其它自定义的 scope
             scope:
               - openid
@@ -50,7 +50,7 @@ spring:
             # client-id 是在 oidc 提供者处配置的客户端ID, 用于登录 provider
             client-id: apollo-portal
             # provider 的名称, 需要和上面配置的 provider 名称保持一致
-            provider: provider-name
+            provider: <fill-in-the-provider-name-here>
             # openid 为 oidc 登录的必须 scope, 此处可以添加其它自定义的 scope
             scope:
               - openid

--- a/docs/zh/development/portal-how-to-implement-user-login-function.md
+++ b/docs/zh/development/portal-how-to-implement-user-login-function.md
@@ -239,6 +239,7 @@ export JAVA_OPTS="$JAVA_OPTS -Dspring.profiles.active=github,ldap"
 解压`apollo-portal-x.x.x-github.zip`后，在`config`目录下创建`application-oidc.yml`，内容参考如下（[样例](https://github.com/apolloconfig/apollo/blob/master/apollo-portal/src/main/config/application-oidc-sample.yml)），相关的内容需要按照具体情况调整：
 
 #### 1.1 最小配置
+
 ```yml
 server:
   # 解析反向代理请求头
@@ -249,20 +250,20 @@ spring:
       client:
         provider:
           # provider-name 是 oidc 提供者的名称, 任意字符均可, registration 的配置需要用到这个名称
-          provider-name:
+          <fill-in-the-provider-name-here>:
             # 必须是 https, oidc 的 issuer-uri
             # 例如 你的 issuer-uri 是 https://host:port/auth/realms/apollo/.well-known/openid-configuration, 那么此处只需要配置 https://host:port/auth/realms/apollo 即可, spring boot 处理的时候会加上 /.well-known/openid-configuration 的后缀
             issuer-uri: https://host:port/auth/realms/apollo
         registration:
           # registration-name 是 oidc 客户端的名称, 任意字符均可, oidc 登录必须配置一个 authorization_code 类型的 registration
-          registration-name:
+          <fill-in-the-registration-name-here>:
             # oidc 登录必须配置一个 authorization_code 类型的 registration
             authorization-grant-type: authorization_code
             client-authentication-method: basic
             # client-id 是在 oidc 提供者处配置的客户端ID, 用于登录 provider
             client-id: apollo-portal
             # provider 的名称, 需要和上面配置的 provider 名称保持一致
-            provider: provider-name
+            provider: <fill-in-the-provider-name-here>
             # openid 为 oidc 登录的必须 scope, 此处可以添加其它自定义的 scope
             scope:
               - openid
@@ -276,6 +277,7 @@ spring:
 #### 1.2 扩展配置
 * 如果 OpenID Connect 登录服务支持 client_credentials 模式, 还可以再配置一个 client_credentials 类型的 registration, 用于 apollo-portal 作为客户端请求其它被 oidc 保护的资源
 * 如果 OpenID Connect 登录服务支持 jwt, 还可以配置 ${spring.security.oauth2.resourceserver.jwt.issuer-uri}, 以支持通过 jwt 访问 apollo-portal
+
 ```yml
 server:
   # 解析反向代理请求头
@@ -286,19 +288,19 @@ spring:
       client:
         provider:
           # provider-name 是 oidc 提供者的名称, 任意字符均可, registration 的配置需要用到这个名称
-          provider-name:
+          <fill-in-the-provider-name-here>:
             # 必须是 https, oidc 的 issuer-uri, 和 jwt 的 issuer-uri 一致的话直接引用即可, 也可以单独设置
             issuer-uri: ${spring.security.oauth2.resourceserver.jwt.issuer-uri}
         registration:
           # registration-name 是 oidc 客户端的名称, 任意字符均可, oidc 登录必须配置一个 authorization_code 类型的 registration
-          registration-name:
+          <fill-in-the-registration-name-here>:
             # oidc 登录必须配置一个 authorization_code 类型的 registration
             authorization-grant-type: authorization_code
             client-authentication-method: basic
             # client-id 是在 oidc 提供者处配置的客户端ID, 用于登录 provider
             client-id: apollo-portal
             # provider 的名称, 需要和上面配置的 provider 名称保持一致
-            provider: provider-name
+            provider: <fill-in-the-provider-name-here>
             # openid 为 oidc 登录的必须 scope, 此处可以添加其它自定义的 scope
             scope:
               - openid
@@ -314,7 +316,7 @@ spring:
             # client-id 是在 oidc 提供者处配置的客户端ID, 用于登录 provider
             client-id: apollo-portal
             # provider 的名称, 需要和上面配置的 provider 名称保持一致
-            provider: provider-name
+            provider: <fill-in-the-provider-name-here>
             # openid 为 oidc 登录的必须 scope, 此处可以添加其它自定义的 scope
             scope:
               - openid
@@ -378,6 +380,7 @@ server {
 
 #### 3.2 检查 application-oidc.yml 配置
 在 `application-oidc.yml` 里必须存在配置项 `server.forward-headers-strategy=framework`
+
 ```yml
 server:
   # 解析反向代理请求头

